src/ergw_aaa_static.erl

```diff
@@ -87,8 +87,8 @@ handle_response(Procedure, #{'Result-Code' := Code} = Avps,
   when Code < 3000 ->
     {Session, Events} = to_session(Procedure, {Session0, Events0}, Avps),
     {ok, Session, Events, State};
-handle_response(_Procedure, #{'Result-Code' := Code}, Session, Events, State) ->
-    {{fail, Code}, Session, [stop | Events], State};
+handle_response(Procedure, #{'Result-Code' := Code}, Session, Events, State) ->
+    {{fail, Code}, Session, [{stop, Procedure} | Events], State};
```

Comment:
couldn't that lead to a situation where we return a `Reason` that has no mapping to a `Termination-Cause` ?

Reply:
Yes, the reason will not be mapped of termination cause, the main goal of this changes - return reason of stop session for handle it in `ergw`.